test/junit/scala/collection/immutable/ArraySeqTest.scala

```diff
@@ -96,6 +96,27 @@ class ArraySeqTest {
     assertEquals(x.hashCode(), y.hashCode())
   }
 
+  @Test
+  def foldInt(): Unit = {
+    val a = ArraySeq(1, 3)
+    assertEquals(a.foldLeft(0)(_ + _), 4)
+    assertEquals(a.foldRight(0)(_ + _), 4)
+  }
+
+  @Test
+  def foldString(): Unit = {
+    val a = ArraySeq("1", "3")
+    assertEquals(a.foldLeft("")(_ + _), "13")
+    assertEquals(a.foldRight("")(_ + _), "13")
```

Comment:
Maybe add a test that actually shows the difference between foldLeft and foldRight? eg

```
scala> ArraySeq("1","2","3").foldRight("0"){case (acc, b) => s"f($acc, $b)"}
val res0: String = f(1, f(2, f(3, 0)))

scala> ArraySeq("1","2","3").foldLeft("0"){case (b, acc) => s"f($b, $acc)"}
val res1: String = f(f(f(0, 1), 2), 3)
```

or just make the accumulator non-empty

Reply:
No worries. How about I can build a list with `(_ :: _)`? Sound ok?

Reply:
Sure!